packages/babel-types/src/definitions/flow.js

```diff
@@ -464,3 +465,80 @@ defineType("Variance", {
 defineType("VoidTypeAnnotation", {
   aliases: ["Flow", "FlowType", "FlowBaseAnnotation"],
 });
+
+// Enums
+defineType("EnumDeclaration", {
+  alises: ["Declaration"],
+  visitor: ["id", "body"],
+  fields: {
+    id: validateType("Identifier"),
+    body: validateType([
+      "EnumBooleanBody",
+      "EnumNumberBody",
+      "EnumStringBody",
+      "EnumSymbolBody",
+    ]),
+  },
+});
+
+defineType("EnumBooleanBody", {
+  visitor: ["members"],
+  fields: {
+    explicit: validate(assertValueType("boolean")),
+    members: validateArrayOfType("EnumBooleanMember"),
+  },
+});
```

Comment:
I would prefer this:

```js
defineType("EnumBody", {
  visitor: ["members"],
  fields: {
    explicit: validate(assertValueType("boolean")),
    type: validate(oneOf("boolean", "string", "symbol", "number"),
    members: validateArrayOfType("EnumBooleanMember"),
  },
});
```

Because I think that most plugins would visit any type of enum.

Reply:
We have designed the AST representation with the goal of making invalid states impossible to represent (as much as possible). For example, a symbol based enum must always be explicit, so an `EnumSymbolBody` does not have an `explicit` field.

Reply:
I would still prefer to have a single `EnumBody` with a `type` attribute and maybe a single `EnumMeber`, but if the rest of the team doesn't think that it is needed maybe an alias would be enough:
```js
defineType("EnumNumberBody", {
  aliases: ["EnumBody"],
  visitor: ["members"],
  fields: {
    explicit: validate(assertValueType("boolean")),
    members: validateArrayOfType("EnumNumberMember"),
  },
});
```

Reply:
I like the idea of using aliases, it works well. I will add an alias for both `EnumBody` and `EnumMember`.